rich/_win32_console.py

```diff
@@ -340,6 +364,12 @@ def __init__(self, file: "IO[str]") -> None:
         default_text = GetConsoleScreenBufferInfo(handle).wAttributes
         self._default_text = default_text
 
+        # Store the current cursor size
+        # @link https://docs.microsoft.com/en-us/windows/console/console-cursor-info-str
+        initial_cursor_info = CONSOLE_CURSOR_INFO()
```

Comment:
I don't think we should be storing the cursor size in the constructor. Consider if the user changes the cursor size after this object has been created (hardly likely, but possible). Now if they call `hide_cursor` or `show_cursor` the cursor size may be incorrect.

We should add this to `hide_cursor` and `show_cursor`. i.e. get the cursor info first, modify the visible flag, and then set the cursor info in a single operation.

Reply:
Done! :slightly_smiling_face:  [08ea255](https://github.com/Textualize/rich/pull/2339/commits/08ea255fe90e8fe8be069c1d420026cb43030389)